File: server/routes/analysis.js
const express = require("express");
const router = express.Router();
const { validateSiteParams, asyncHandler } = require("../middleware/validate");
const { success } = require("../utils/response");

const { proximitySummary, nearestNeighbours } = require("../services/proximityService");
const { landUseBreakdown, buildingStats } = require("../services/landUseService");
const { transportStats } = require("../services/transportService");
const { amenityScore } = require("../services/amenityService");
const { environmentalScan } = require("../services/environmentService");
const { connectivityAnalysis } = require("../services/connectivityService");
const { riskAssessment } = require("../services/riskService");
const { footfallAnalysis } = require("../services/footfallService");

// ────────────────────────────────────────────────────
// Analysis Endpoints (All use asyncHandler + success())
// ────────────────────────────────────────────────────

router.get("/proximity", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await proximitySummary(req.siteParams);
  res.json(success(data));
}));

router.get("/footfall", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await footfallAnalysis(req.siteParams);
  res.json(success(data));
}));

router.get("/neighbours", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await nearestNeighbours(req.siteParams);
  res.json(success(data));
}));

router.get("/landuse", validateSiteParams, asyncHandler(async (req, res) => {
  const [breakdown, buildings] = await Promise.all([
    landUseBreakdown(req.siteParams),
    buildingStats(req.siteParams),
  ]);
  res.json(success({ breakdown, buildings }));
}));

router.get("/transport", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await transportStats(req.siteParams);
  res.json(success(data));
}));

router.get("/amenity-score", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await amenityScore(req.siteParams);
  res.json(success(data));
}));

router.get("/environment", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await environmentalScan(req.siteParams);
  res.json(success(data));
}));

router.get("/connectivity", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await connectivityAnalysis(req.siteParams);
  res.json(success(data));
}));

router.get("/risk", validateSiteParams, asyncHandler(async (req, res) => {
  const data = await riskAssessment(req.siteParams);
  res.json(success(data));
}));

router.get("/full", validateSiteParams, asyncHandler(async (req, res) => {
  const [
    proximity, neighbours, landuse, buildings,
    transport, amenity, environment, connectivity, risk, footfall
  ] = await Promise.all([
    proximitySummary(req.siteParams),
    nearestNeighbours(req.siteParams),
    landUseBreakdown(req.siteParams),
    buildingStats(req.siteParams),
    transportStats(req.siteParams),
    amenityScore(req.siteParams),
    environmentalScan(req.siteParams),
    connectivityAnalysis(req.siteParams),
    riskAssessment(req.siteParams),
    footfallAnalysis(req.siteParams),
  ]);

  res.json(success({
    params: req.siteParams,
    data: {
      proximity,
      neighbours,
      landuse: { breakdown: landuse, buildings },
      transport,
      amenity,
      environment,
      connectivity,
      risk,
      footfall,
    },
  }));
}));

module.exports = router;

File: backend-py/app/routes/analysis.py
"""Port of server/routes/analysis.js — all endpoints require auth."""
from fastapi import APIRouter, Depends

from ..auth import require_auth
from ..common import success, validate_site_params
from ..util import gather
from ..services.proximity import proximity_summary, nearest_neighbours
from ..services.landuse import land_use_breakdown, building_stats
from ..services.transport import transport_stats
from ..services.amenity import amenity_score
from ..services.environment import environmental_scan
from ..services.connectivity import connectivity_analysis
from ..services.risk import risk_assessment
from ..services.footfall import footfall_analysis

router = APIRouter(prefix="/api/analysis", dependencies=[Depends(require_auth)])


def _params(lat=None, lng=None, radius=None):
    return validate_site_params(lat, lng, radius)


@router.get("/proximity")
def proximity(lat: str = None, lng: str = None, radius: str = None):
    return success(proximity_summary(_params(lat, lng, radius)))


@router.get("/footfall")
def footfall(lat: str = None, lng: str = None, radius: str = None):
    return success(footfall_analysis(_params(lat, lng, radius)))


@router.get("/neighbours")
def neighbours(lat: str = None, lng: str = None, radius: str = None):
    return success(nearest_neighbours(_params(lat, lng, radius)))


@router.get("/landuse")
def landuse(lat: str = None, lng: str = None, radius: str = None):
    p = _params(lat, lng, radius)
    breakdown, buildings = gather(lambda: land_use_breakdown(p), lambda: building_stats(p))
    return success({"breakdown": breakdown, "buildings": buildings})


@router.get("/transport")
def transport(lat: str = None, lng: str = None, radius: str = None):
    return success(transport_stats(_params(lat, lng, radius)))


@router.get("/amenity-score")
def amenity(lat: str = None, lng: str = None, radius: str = None):
    return success(amenity_score(_params(lat, lng, radius)))


@router.get("/environment")
def environment(lat: str = None, lng: str = None, radius: str = None):
    return success(environmental_scan(_params(lat, lng, radius)))


@router.get("/connectivity")
def connectivity(lat: str = None, lng: str = None, radius: str = None):
    return success(connectivity_analysis(_params(lat, lng, radius)))


@router.get("/risk")
def risk(lat: str = None, lng: str = None, radius: str = None):
    return success(risk_assessment(_params(lat, lng, radius)))


@router.get("/full")
def full(lat: str = None, lng: str = None, radius: str = None):
    p = _params(lat, lng, radius)
    (proximity_r, neighbours_r, landuse_r, buildings_r, transport_r,
     amenity_r, environment_r, connectivity_r, risk_r, footfall_r) = gather(
        lambda: proximity_summary(p),
        lambda: nearest_neighbours(p),
        lambda: land_use_breakdown(p),
        lambda: building_stats(p),
        lambda: transport_stats(p),
        lambda: amenity_score(p),
        lambda: environmental_scan(p),
        lambda: connectivity_analysis(p),
        lambda: risk_assessment(p),
        lambda: footfall_analysis(p),
    )
    return success({
        "params": p,
        "data": {
            "proximity": proximity_r,
            "neighbours": neighbours_r,
            "landuse": {"breakdown": landuse_r, "buildings": buildings_r},
            "transport": transport_r,
            "amenity": amenity_r,
            "environment": environment_r,
            "connectivity": connectivity_r,
            "risk": risk_r,
            "footfall": footfall_r,
        },
    })
